packages/xsnap/src/build.js

```diff
@@ -297,76 +287,77 @@ async function main(args, { env, stdout, spawn, fs, os }) {
   await null;
 
   const osType = os.type();
-  const platform = {
-    Linux: 'lin',
-    Darwin: 'mac',
-    // Windows_NT: 'win', // One can dream.
-  }[osType];
-  if (platform === undefined) {
-    throw Error(`xsnap does not support platform ${osType}`);
+  const platform = ModdableSDK.platforms[osType];
+  if (!platform) {
+    throw Error(`xsnap does not support OS ${osType}`);
   }
 
-  // If this is a working copy of xsnap in a checkout of agoric-sdk, we need to
-  // either clone or update submodules.
-  // Otherwise, we are running from an extracted npm tarball and we should not
-  // attempt to update Git submodules and should make the binary from the
-  // published source.
-  //
-  // These steps will avoid rebuilding native xsnap in the common case for end
-  // users.
-  //
-  //                    ||      | X    || git
-  //                    || X    | X    || make
-  //                    || ---- | ---- || ----
-  // | bin | src | .git || pack | work ||
-  // | --- | --- | ---- || ---- | ---- ||
-  // |     |     |      ||      | X    ||
-  // |     |     | X    ||      |      ||
-  // |     | X   |      || X    |      ||
-  // |     | X   | X    ||      | X    ||
-  // | X   |     |      ||      |      ||
-  // | X   |     | X    ||      |      ||
-  // | X   | X   |      || X    |      ||
-  // | X   | X   | X    ||      | X    ||
-  //
-  // We build both release and debug, so checking for one should suffice.
```

Comment:
Looks like we lose this lore, which we might need to reconstruct if we need to come up with different logic. Is it more confusing in this form than the commented logic?

Reply:
> Is it more confusing in this form than the commented logic?

For me, yes, especially with the actual logic being rather simple (`isWorkingCopy := hasGit || (!hasSource && !hasBin)`, `isWorkingCopy ⇒ updateSubmodules`, `hasSource || isWorkingCopy ⇒ buildXsnap` with `force := !hasBin || <check-version does not pass>`). But I can try restoring and correcting this if you prefer.